decidim-stats/app/queries/decidim/stats/actions/endorsement.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Stats
+    module Actions
+      class Endorsement
+        def initialize(component, performers)
+          @component = component
+          @performers = performers
+        end
+
+        def query
+          return [] unless proposals_manifest?
+
+          @query ||=
+            query_base
+            .where(decidim_user_group_id: nil)
+            .where(author: performers)
+            .pluck(:decidim_author_id)
+            .uniq
+        end
+
+        private
+
+        attr_reader :component, :performers
+
+        def component_manifest
+          component.manifest
+        end
+
+        def query_base
+          Decidim::Endorsement.where(resource: proposals)
+        rescue NameError
+          Decidim::Proposals::ProposalEndorsement
```

Comment:
woww!!! :+1:

Reply:
Saw your PR and thought it was better handling it now than having to send another PR in the future 😄